### DISCUSSION TITLE
Switch to Vanilla Extract

### DISCUSSION THREAD

**Discussion by eliasljubec** [gluestack/gluestack-ui#949] [Feature Request]
### Description

Hello.
I saw that you are using styled-components under the hood for glustack-ui & since the project is fairly new, I would perhaps suggest that you switch to vanilla-extract since the problem with styled-components is that they recalculate styles on runtime, that means it needs to build up the stylesheet and insert it into the document, every time a component is rendered. For small applications this is not a problem, but for larger applications it affects performance.

### Problem Statement

Vanilla Extract generates the CSS at build time, which improves performance compared to styled-components, which evaluate it at runtime.

### Proposed Solution or API

https://vanilla-extract.style/

**Comment by Viraj-10** [comment]
Hey @eljubec, Currently in `gluestack-style` we have bare minimum logic on run time. We have all our styling calculation at boot time and with `babel-plugin` it's at build time. We just inject style id's when we get the `dom` in case of web.
But we are considering this option to extract css as well.
